runtime/lib/ttnn/program.cpp

```diff
@@ -313,6 +313,61 @@ run(::tt::target::ttnn::ReductionOp const *op, ::ttnn::Device &device,
   }
 }
 
+template <int32_t Rank>
+static std::array<int32_t, Rank>
+vectorToArray(const std::vector<int32_t> &vec) {
+  if (vec.size() != Rank) {
+    throw std::invalid_argument("Vector size does not match array size");
+  }
+  std::array<int32_t, Rank> arr;
+  std::copy(vec.begin(), vec.end(), arr.begin());
+  return arr;
+}
+
+template <int32_t Rank>
+static ::ttnn::Tensor invoke_reshape(const ::ttnn::Tensor &tensor,
+                                     const std::vector<int32_t> &shape) {
+  return ::ttnn::reshape(tensor, vectorToArray<Rank>(shape));
+}
+
+static void
+run(::tt::target::ttnn::ReshapeOp const *op, ::ttnn::Device &device,
+    std::unordered_map<std::uint32_t, ::ttnn::Tensor *> &liveTensors,
+    std::list<::ttnn::Tensor> &tensorPool) {
+  auto &in = *liveTensors.at(op->in()->global_id());
+  const auto *shape_arg_fb_ptr = op->shape();
+  std::vector<int32_t> shape(shape_arg_fb_ptr->begin(),
+                             shape_arg_fb_ptr->end());
+
+  constexpr int32_t Rank1 = 1;
+  constexpr int32_t Rank2 = 2;
+  constexpr int32_t Rank3 = 3;
+  constexpr int32_t Rank4 = 4;
+  constexpr int32_t Rank5 = 5;
+
+  switch (shape_arg_fb_ptr->size()) {
+  case Rank1:
+    tensorPool.push_back(invoke_reshape<Rank1>(in, shape));
```

Comment:
Can this whole switch case be simplified into something like:
```
tensorPool.push_back(invoke_reshape<shape_arg_fb_ptr->size()>(in, shape));
```

Reply:
No, template parameters need to be known at compile time...